authz/sdk_server_interceptors_test.go

```diff
@@ -43,14 +67,51 @@ func TestNewStatic(t *testing.T) {
 					}
 				]
 			}`,
-			wantErr: false,
 		},
 	}
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
-			if _, err := authz.NewStatic(test.authzPolicy); (err != nil) != test.wantErr {
+			if _, err := authz.NewStatic(test.authzPolicy); (err != nil) && (err.Error() != test.wantErr) {
 				t.Fatalf("NewStatic(%v) returned err: %v, want err: %v", test.authzPolicy, err, test.wantErr)
 			}
 		})
 	}
 }
+
+func TestNewFileWatcher(t *testing.T) {
+	tests := map[string]struct {
+		authzPolicy     string
+		refreshDuration time.Duration
+		wantErr         string
+	}{
+		"InvalidRefreshDurationFailsToCreateInterceptor": {
+			refreshDuration: time.Duration(0),
+			wantErr:         "requires refresh interval(0s) greater than 0s",
+		},
+		"InvalidPolicyFailsToCreateInterceptor": {
+			authzPolicy:     `{}`,
+			refreshDuration: time.Duration(1),
+			wantErr:         `"name" is not present`,
+		},
+		"ValidPolicyCreatesInterceptor": {
+			authzPolicy: `{
+				"name": "authz",
+				"allow_rules":
+				[
+					{
+						"name": "allow_all"
+					}
+				]
+			}`,
+			refreshDuration: time.Duration(1),
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			dir := createTmpPolicyFile(t, name+"*", []byte(test.authzPolicy))
+			if _, err := authz.NewFileWatcher(path.Join(dir, "policy.json"), test.refreshDuration); (err != nil) && (err.Error() != test.wantErr) {
```

Comment:
Same as above.  This is why the test is passing even though the constructor does not error on an invalid policy file.

Reply:
Done.